In [1]:
# TensorFlow and tf.keras
import tensorflow as tf
from tensorflow import keras

# Helper libraries
import numpy as np
import matplotlib.pyplot as plt

print(tf.__version__)

/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters


1.9.0


Load and prepare the [MNIST](http://yann.lecun.com/exdb/mnist/) dataset. Convert the samples from integers to floating-point numbers:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 15s 1us/step


Build the `tf.keras` model by stacking layers. Select an optimizer and loss function used for training:

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate model:

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Epoch 1/5
60000/60000 [==============================] - 21s 358us/step - loss: 0.2203 - acc: 0.9345
Epoch 2/5
60000/60000 [==============================] - 22s 369us/step - loss: 0.0969 - acc: 0.9700
Epoch 3/5
60000/60000 [==============================] - 18s 306us/step - loss: 0.0692 - acc: 0.9782
Epoch 4/5
60000/60000 [==============================] - 19s 309us/step - loss: 0.0526 - acc: 0.9830
Epoch 5/5
10000/10000 [==============================] - 1s 61us/step


[0.06071833839677565, 0.9815]

You’ve now trained an image classifier with ~98% accuracy on this dataset. See Get Started with TensorFlow to learn more.